--- specifications/introspection-0.0.2.gemspec.orig
+++ specifications/introspection-0.0.2.gemspec
@@ -20,16 +20,13 @@
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
       s.add_runtime_dependency(%q<metaclass>, ["~> 0.0.1"])
-      s.add_runtime_dependency(%q<instantiator>, ["~> 0.0.3"])
       s.add_development_dependency(%q<rake>, [">= 0"])
     else
       s.add_dependency(%q<metaclass>, ["~> 0.0.1"])
-      s.add_dependency(%q<instantiator>, ["~> 0.0.3"])
       s.add_dependency(%q<rake>, [">= 0"])
     end
   else
     s.add_dependency(%q<metaclass>, ["~> 0.0.1"])
-    s.add_dependency(%q<instantiator>, ["~> 0.0.3"])
     s.add_dependency(%q<rake>, [">= 0"])
   end
 end
