--- rubygems-1.8.5/lib/rubygems.rb.all_load_paths	2011-06-25 13:57:51.000000000 +0900
+++ rubygems-1.8.5/lib/rubygems.rb	2011-06-25 15:03:26.000000000 +0900
@@ -258,7 +258,8 @@
 
     Gem.path.each do |gemdir|
       each_load_path all_partials(gemdir) do |load_path|
-        result << gemdir.add(load_path).expand_path
+#        result << gemdir.add(load_path).expand_path
+        result << load_path
       end
     end
 
