--- ruby-1.8.7-p249/lib/mkmf.rb.orig	2010-06-15 12:04:06.000000000 -0400
+++ ruby-1.8.7-p249/lib/mkmf.rb	2010-06-15 12:05:05.000000000 -0400
@@ -319,7 +319,7 @@ def link_command(ldflags, opt="", libpat
                               'LDFLAGS' => "#$LDFLAGS #{ldflags}",
                               'LIBPATH' => libpathflag(libpath),
                               'LOCAL_LIBS' => "#$LOCAL_LIBS #$libs",
-                              'LIBS' => "#$LIBRUBYARG_STATIC #{opt} #$LIBS")
+                              'LIBS' => "#$LIBRUBYARG_SHARED #{opt} #$LIBS")
   Config::expand(TRY_LINK.dup, conf)
 end
 
