--- a/specifications/activesupport-3.2.8.gemspec.orig	2012-07-18 14:52:50.000000000 +0200
+++ b/specifications/activesupport-3.2.8.gemspec	2012-07-18 14:56:49.367668024 +0200
@@ -20,13 +20,16 @@
     s.specification_version = 3
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
+      s.add_runtime_dependency(%q<bigdecimal>, [">= 0"])
       s.add_runtime_dependency(%q<i18n>, ["~> 0.6"])
       s.add_runtime_dependency(%q<multi_json>, ["~> 1.0"])
     else
+      s.add_dependency(%q<bigdecimal>, [">= 0"])
       s.add_dependency(%q<i18n>, ["~> 0.6"])
       s.add_dependency(%q<multi_json>, ["~> 1.0"])
     end
   else
+    s.add_dependency(%q<bigdecimal>, [">= 0"])
     s.add_dependency(%q<i18n>, ["~> 0.6"])
     s.add_dependency(%q<multi_json>, ["~> 1.0"])
   end
