From a6c1ee7c58b145a361bb614f96ef575091e70512 Mon Sep 17 00:00:00 2001
From: Dominic Cleal <dcleal@redhat.com>
Date: Wed, 19 Nov 2014 10:06:44 +0000
Subject: [PATCH 2/4] Remove fog-radosgw dependency

---
 fog.gemspec | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fog.gemspec b/fog.gemspec
index 5aea4d8..9fa3146 100644
--- a/fog.gemspec
+++ b/fog.gemspec
@@ -50,7 +50,6 @@ Gem::Specification.new do |s|
   # Modular providers
   s.add_dependency("fog-softlayer")
   s.add_dependency("fog-sakuracloud", ">= 0.0.4")
-  s.add_dependency("fog-radosgw", ">=0.0.2")
 
   ## List your development dependencies here. Development dependencies are
   ## those that are only needed during development
-- 
1.9.3

