diff --git a/examples/pdf_table.rb b/examples/pdf_table.rb
index fb14f87..4be9e86 100644
--- a/examples/pdf_table.rb
+++ b/examples/pdf_table.rb
@@ -6,7 +6,7 @@ class Document < Ruport::Controller
   stage :body

   def setup
-    self.data = Ruport::Table(
+    self.data = Ruport::Data::Table.new(
       :column_names => %w(Make Model Year),
       :data => [
         %w(Nissan Skyline 1989),
diff --git a/examples/pdf_table_simple.rb b/examples/pdf_table_simple.rb
index b4f2640..024dd91 100644
--- a/examples/pdf_table_simple.rb
+++ b/examples/pdf_table_simple.rb
@@ -2,7 +2,7 @@ $: << File.join(File.dirname(__FILE__), '..', 'lib')
 require "ruport"

 # Quick and simple example using prawn 0.9.0 with to_prawn_pdf.
-table = Ruport::Table(
+table = Ruport::Data::Table.new(
           :column_names => %w(Make Model Year),
           :data => [
             %w(Nissan Skyline 1989),
diff --git a/lib/ruport/controller.rb b/lib/ruport/controller.rb
index ab9f742..3906881 100644
--- a/lib/ruport/controller.rb
+++ b/lib/ruport/controller.rb
@@ -219,7 +219,7 @@ class Ruport::Controller
      { :html => Ruport::Formatter::HTML,
        :csv  => Ruport::Formatter::CSV,
        :pdf  => Ruport::Formatter::PDF,
-       :pdf_prawn => Ruport::Formatter::PrawnPDF,
+       :prawn_pdf => Ruport::Formatter::PrawnPDF,
        :text => Ruport::Formatter::Text }
     end

diff --git a/lib/ruport/formatter/pdf.rb b/lib/ruport/formatter/pdf.rb
index 7832044..97f5eef 100644
--- a/lib/ruport/formatter/pdf.rb
+++ b/lib/ruport/formatter/pdf.rb
@@ -276,7 +276,6 @@ module Ruport
     #
     # http://stonecode.svnrepository.com/ruport/trac.cgi/wiki/PdfWriterQuickRef
     def draw_table(table_data, format_opts={})
-
       m = "PDF Formatter requires column_names to be defined"
       raise FormatterError, m if table_data.column_names.empty?

@@ -393,7 +392,6 @@ module Ruport
     private

     def apply_pdf_table_column_opts(table,table_data,format_opts)
-
       column_opts = format_opts.delete(:column_options)

       if column_opts
diff --git a/lib/ruport/formatter/prawn_pdf.rb b/lib/ruport/formatter/prawn_pdf.rb
index 8a2be6b..b1d9a16 100644
--- a/lib/ruport/formatter/prawn_pdf.rb
+++ b/lib/ruport/formatter/prawn_pdf.rb
@@ -17,12 +17,10 @@ module Ruport

     def pdf
       @pdf ||= (options.formatter ||
-        ::Prawn::Document.new(options[:pdf_format] || {:page_size => "LETTER",
-          :page_layout => :portrait} ))
+        ::Prawn::Document.new(options[:pdf_format] || {} ))
     end

     def draw_table(table, format_opts={})
-
       m = "PDF Formatter requires column_names to be defined"
       raise FormatterError, m if table.column_names.empty?
