diff --git a/ruport.gemspec b/ruport.gemspec
index 7ba3dd7..a90dc89 100644
--- a/ruport.gemspec
+++ b/ruport.gemspec
@@ -152,16 +152,16 @@ Gem::Specification.new do |s|
     s.specification_version = 3
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
-      s.add_runtime_dependency(%q<fastercsv>, [">= 0"])
+      s.add_runtime_dependency(%q<fastercsv>, [">= 0"]) if RUBY_VERSION < "1.9"
       s.add_runtime_dependency(%q<pdf-writer>, ["= 1.1.8"])
       s.add_runtime_dependency(%q<prawn>, [">= 0.12.0"])
     else
-      s.add_dependency(%q<fastercsv>, [">= 0"])
+      s.add_dependency(%q<fastercsv>, [">= 0"]) if RUBY_VERSION < "1.9"
       s.add_dependency(%q<pdf-writer>, ["= 1.1.8"])
       s.add_runtime_dependency(%q<prawn>, [">= 0.12.0"])
     end
   else
-    s.add_dependency(%q<fastercsv>, [">= 0"])
+    s.add_dependency(%q<fastercsv>, [">= 0"]) if RUBY_VERSION < "1.9"
     s.add_dependency(%q<pdf-writer>, ["= 1.1.8"])
     s.add_runtime_dependency(%q<prawn>, [">= 0.12.0"])
   end
