--- lib/hoe.rb.manifest	1970-01-01 09:00:00.000000000 +0900
+++ lib/hoe.rb	2011-08-28 04:15:47.000000000 +0900
@@ -424,7 +424,7 @@

       manifest = File.read_utf("Manifest.txt").split(/\r?\n\r?/) rescue nil

-      abort [
+      warn [
              "Manifest is missing or couldn't be read.",
              "The Manifest is kind of a big deal.",
              "Maybe you're using a gem packaged by a linux project.",
