diff -uNr foreman/Gemfile foreman-nogit/Gemfile
--- foreman/Gemfile	2012-06-21 10:47:27.318016653 -0400
+++ foreman-nogit/Gemfile	2012-06-21 10:48:02.482250207 -0400
@@ -5,7 +5,7 @@
 gem 'json'
 gem 'rest-client', :require => 'rest_client'
 gem "audited-activerecord", "3.0.0.rc1"
-gem "has_many_polymorphs", :git => "https://github.com/jystewart/has_many_polymorphs.git", :ref => '03429a61e511f394e9f96af0c8998268ca99d42b'
+gem "has_many_polymorphs"
 gem "will_paginate", "~> 3.0.2"
 gem "ancestry", "~> 1.2.4"
 gem 'scoped_search', '>= 2.4.0'
