--- foreman.spec-old	2012-08-05 14:06:24.972065082 -0400
+++ foreman.spec	2012-08-05 14:07:05.213569479 -0400
@@ -45,6 +45,9 @@
 Requires: rubygem(rake) >= 0.9.2.2
 Requires: rubygem(ruby_parser) >= 2.3.1
 Requires: rubygem(audited-activerecord) >= 3.0.0
+Requires: rubygem(apipie-rails) >= 0.0.7
+Requires: rubygem(oauth)
+Requires: rubygem(rabl)
 Provides: %{name}-%{version}-%{release}
 #Packager:   Ohad Levy <ohadlevy@gmail.com>
 
