diff -uNr foreman-proxy-orig/bin/smart-proxy foreman-proxy/bin/smart-proxy
--- foreman-proxy-orig/bin/smart-proxy	2012-06-21 15:52:48.585997827 -0400
+++ foreman-proxy/bin/smart-proxy	2012-06-21 15:53:47.910000051 -0400
@@ -63,7 +63,7 @@
 end
 
 if ARGV[0] == "--service"
-  raise "The service flag is used only in a windows environment" unless PLATFORM =~ /mingw/
+  raise "The service flag is used only in a windows environment" unless RUBY_PLATFORM =~ /mingw/
     begin
       require 'win32/daemon'
       include Win32
diff -uNr foreman-proxy-orig/extra/query.rb foreman-proxy/extra/query.rb
--- foreman-proxy-orig/extra/query.rb	2012-06-21 15:52:48.583997676 -0400
+++ foreman-proxy/extra/query.rb	2012-06-21 15:53:47.911000046 -0400
@@ -65,7 +65,7 @@
 end
 
 unless key and cert and ca
-  if PLATFORM =~ /mingw/
+  if RUBY_PLATFORM =~ /mingw/
     origin = Pathname.new(__FILE__).dirname.parent.join "config"
     key    = origin.join "private.pem" unless key
     cert   = origin.join "signed.pem"  unless cert
diff -uNr foreman-proxy-orig/extra/register-service.rb foreman-proxy/extra/register-service.rb
--- foreman-proxy-orig/extra/register-service.rb	2012-06-21 15:52:48.583997676 -0400
+++ foreman-proxy/extra/register-service.rb	2012-06-21 15:53:47.911000046 -0400
@@ -1,5 +1,5 @@
 #!c:\ruby187\bin\ruby
-if PLATFORM !~ /mingw/
+if RUBY_PLATFORM !~ /mingw/
   puts "To install this service on Unix please create a startup script"
 else
   require 'rubygems'
Binary files foreman-proxy-orig/.git/index and foreman-proxy/.git/index differ
diff -uNr foreman-proxy-orig/.git/logs/HEAD foreman-proxy/.git/logs/HEAD
--- foreman-proxy-orig/.git/logs/HEAD	2012-06-21 15:52:48.581997515 -0400
+++ foreman-proxy/.git/logs/HEAD	2012-06-21 15:53:48.047000049 -0400
@@ -1 +1,2 @@
 0000000000000000000000000000000000000000 fcf097c2ef3d8eb6295293efec44b2e466c5c476 Jason Montleon <jmontleo@redhat.com> 1340306947 -0400	clone: from git://github.com/theforeman/smart-proxy.git
+fcf097c2ef3d8eb6295293efec44b2e466c5c476 438a05092fbada6d0b403668adf14ebbc0790956 Jason Montleon <jmontleo@redhat.com> 1340308428 -0400	am: Renamed global variable PLATFORM to RUBY_PLATFORM
diff -uNr foreman-proxy-orig/.git/logs/refs/heads/develop foreman-proxy/.git/logs/refs/heads/develop
--- foreman-proxy-orig/.git/logs/refs/heads/develop	2012-06-21 15:52:48.580997423 -0400
+++ foreman-proxy/.git/logs/refs/heads/develop	2012-06-21 15:53:48.047000049 -0400
@@ -1 +1,2 @@
 0000000000000000000000000000000000000000 fcf097c2ef3d8eb6295293efec44b2e466c5c476 Jason Montleon <jmontleo@redhat.com> 1340306947 -0400	clone: from git://github.com/theforeman/smart-proxy.git
+fcf097c2ef3d8eb6295293efec44b2e466c5c476 438a05092fbada6d0b403668adf14ebbc0790956 Jason Montleon <jmontleo@redhat.com> 1340308428 -0400	am: Renamed global variable PLATFORM to RUBY_PLATFORM
Binary files foreman-proxy-orig/.git/objects/17/667462aae09a73c4243d5525d63994a3ec8d27 and foreman-proxy/.git/objects/17/667462aae09a73c4243d5525d63994a3ec8d27 differ
Binary files foreman-proxy-orig/.git/objects/3d/7e680e90685b133eaec39ceba77793259db485 and foreman-proxy/.git/objects/3d/7e680e90685b133eaec39ceba77793259db485 differ
Binary files foreman-proxy-orig/.git/objects/43/8a05092fbada6d0b403668adf14ebbc0790956 and foreman-proxy/.git/objects/43/8a05092fbada6d0b403668adf14ebbc0790956 differ
Binary files foreman-proxy-orig/.git/objects/44/2af28bef6d81275672ec2d796ac5c6f7240af7 and foreman-proxy/.git/objects/44/2af28bef6d81275672ec2d796ac5c6f7240af7 differ
Binary files foreman-proxy-orig/.git/objects/4b/020bcfc451dc43779b18955b73969f23ed48af and foreman-proxy/.git/objects/4b/020bcfc451dc43779b18955b73969f23ed48af differ
Binary files foreman-proxy-orig/.git/objects/51/fde6f1ea0ae68be697ca5cd117a8cda2a82e63 and foreman-proxy/.git/objects/51/fde6f1ea0ae68be697ca5cd117a8cda2a82e63 differ
Binary files foreman-proxy-orig/.git/objects/74/f189a3187f829fbccf7a076107beb59791415f and foreman-proxy/.git/objects/74/f189a3187f829fbccf7a076107beb59791415f differ
Binary files foreman-proxy-orig/.git/objects/96/a8a19a0eb6d36525d42f3cba37248f1c7c1516 and foreman-proxy/.git/objects/96/a8a19a0eb6d36525d42f3cba37248f1c7c1516 differ
Binary files foreman-proxy-orig/.git/objects/9f/41cbd8b4ef8a70e16b3c03f192035395e81d75 and foreman-proxy/.git/objects/9f/41cbd8b4ef8a70e16b3c03f192035395e81d75 differ
Binary files foreman-proxy-orig/.git/objects/a4/d1d6dd65c5498ae27e68dcdd1bb74a41d39314 and foreman-proxy/.git/objects/a4/d1d6dd65c5498ae27e68dcdd1bb74a41d39314 differ
Binary files foreman-proxy-orig/.git/objects/c5/2f936ffe089ea4dd2bd2a039acb309ad13a764 and foreman-proxy/.git/objects/c5/2f936ffe089ea4dd2bd2a039acb309ad13a764 differ
Binary files foreman-proxy-orig/.git/objects/cb/07326a32e28a72713935601e8b0bfe87f76c9e and foreman-proxy/.git/objects/cb/07326a32e28a72713935601e8b0bfe87f76c9e differ
Binary files foreman-proxy-orig/.git/objects/cd/4a7e260a54b2e21cf99f740a23ff5c2407c8a0 and foreman-proxy/.git/objects/cd/4a7e260a54b2e21cf99f740a23ff5c2407c8a0 differ
Binary files foreman-proxy-orig/.git/objects/ee/dcff035ae0543079f56c13fb63b16c4cdc8016 and foreman-proxy/.git/objects/ee/dcff035ae0543079f56c13fb63b16c4cdc8016 differ
diff -uNr foreman-proxy-orig/.git/ORIG_HEAD foreman-proxy/.git/ORIG_HEAD
--- foreman-proxy-orig/.git/ORIG_HEAD	1969-12-31 19:00:00.000000000 -0500
+++ foreman-proxy/.git/ORIG_HEAD	2012-06-21 15:53:47.202000176 -0400
@@ -0,0 +1 @@
+fcf097c2ef3d8eb6295293efec44b2e466c5c476
diff -uNr foreman-proxy-orig/.git/refs/heads/develop foreman-proxy/.git/refs/heads/develop
--- foreman-proxy-orig/.git/refs/heads/develop	2012-06-21 15:52:48.581997515 -0400
+++ foreman-proxy/.git/refs/heads/develop	2012-06-21 15:53:48.047000049 -0400
@@ -1 +1 @@
-fcf097c2ef3d8eb6295293efec44b2e466c5c476
+438a05092fbada6d0b403668adf14ebbc0790956
diff -uNr foreman-proxy-orig/lib/proxy/dhcp/subnet.rb foreman-proxy/lib/proxy/dhcp/subnet.rb
--- foreman-proxy-orig/lib/proxy/dhcp/subnet.rb	2012-06-21 15:52:48.578997200 -0400
+++ foreman-proxy/lib/proxy/dhcp/subnet.rb	2012-06-21 15:53:47.911000046 -0400
@@ -193,7 +193,7 @@
     end
 
     def privileged_user
-      (PLATFORM =~ /linux/i and Process.uid == 0) or PLATFORM =~ /mingw/
+      (RUBY_PLATFORM =~ /linux/i and Process.uid == 0) or PLATFORM =~ /mingw/
     end
   end
 end
diff -uNr foreman-proxy-orig/lib/proxy/settings.rb foreman-proxy/lib/proxy/settings.rb
--- foreman-proxy-orig/lib/proxy/settings.rb	2012-06-21 15:52:48.578997200 -0400
+++ foreman-proxy/lib/proxy/settings.rb	2012-06-21 15:53:47.912000044 -0400
@@ -10,7 +10,7 @@
 end
 
 settings = YAML.load(raw_config)
-if PLATFORM =~ /mingw/
+if RUBY_PLATFORM =~ /mingw/
   settings.delete :puppetca if settings.has_key? :puppetca
   settings.delete :puppet   if settings.has_key? :puppet
   settings[:x86_64] = File.exist?('c:\windows\sysnative\cmd.exe')
diff -uNr foreman-proxy-orig/lib/sinatra-patch.rb foreman-proxy/lib/sinatra-patch.rb
--- foreman-proxy-orig/lib/sinatra-patch.rb	2012-06-21 15:52:48.578997200 -0400
+++ foreman-proxy/lib/sinatra-patch.rb	2012-06-21 15:53:47.912000044 -0400
@@ -1,7 +1,7 @@
 require "sinatra/base"
 require "openssl"
 require "webrick/https"
-require "daemon" unless PLATFORM =~/mingw/
+require "daemon" unless RUBY_PLATFORM =~/mingw/
 module Sinatra
   class Base
     # Run the Sinatra app as a self-hosted server using
@@ -18,7 +18,7 @@
 
       FileUtils.mkdir_p(File.join(APP_ROOT, 'tmp/pids'))
 
-      if SETTINGS.daemon and PLATFORM !~ /mingw/
+      if SETTINGS.daemon and RUBY_PLATFORM !~ /mingw/
         Process.daemon(true)
         if SETTINGS.daemon_pid.nil?
           pid = "#{APP_ROOT}/tmp/pids/server.pid"
diff -uNr foreman-proxy-orig/Rakefile foreman-proxy/Rakefile
--- foreman-proxy-orig/Rakefile	2012-06-21 15:52:48.578997200 -0400
+++ foreman-proxy/Rakefile	2012-06-21 15:53:47.910000051 -0400
@@ -13,7 +13,7 @@
   t.libs << 'lib'
   t.libs << 'test'
   files = FileList['test/**/*_test.rb']
-  if PLATFORM =~ /mingw/
+  if RUBY_PLATFORM =~ /mingw/
     files = FileList['test/**/server_ms_test*']
   else
     files = FileList['test/**/*_test.rb'].delete_if{|f| f =~ /_ms_/}
