commit c8f3df3369f0d5b32dbc57cf3fd575b50d106563
Author: Guillermo Iguaran <guilleiguaran@gmail.com>
Date:   Wed Apr 4 10:18:20 2012 -0500

    Fix asset test

diff --git a/test/assets_test.rb b/test/assets_test.rb
index 9249572..6b9104b 100644
--- a/test/assets_test.rb
+++ b/test/assets_test.rb
@@ -25,7 +25,7 @@ class AssetsTest < ActiveSupport::TestCase
     @app.assets["coffee-script"].write_to("#{tmp_path}/coffee-script.js")
 
     assert_match "/lib/assets/javascripts/coffee-script.js.erb", @app.assets["coffee-script"].pathname.to_s
-    assert_match "this.CoffeeScript", File.open("#{tmp_path}/coffee-script.js").read
+    assert_match "CoffeeScript Compiler", File.open("#{tmp_path}/coffee-script.js").read
   end
 
   def tmp_path
