commit ff097e8722056dfef6ac4523d406bdbca6eae87d
Author: HannesG <hag@informatik.uni-kiel.de>
Date:   Sun Jan 8 21:15:26 2012 +0100

    adjusted specs to RDoc >= 3.10 ( which outputs header ids )

diff --git a/test/tilt_rdoctemplate_test.rb b/test/tilt_rdoctemplate_test.rb
index 309a048..33d8e20 100644
--- a/test/tilt_rdoctemplate_test.rb
+++ b/test/tilt_rdoctemplate_test.rb
@@ -11,12 +11,12 @@ begin
 
     test "preparing and evaluating the template with #render" do
       template = Tilt::RDocTemplate.new { |t| "= Hello World!" }
-      assert_equal "<h1>Hello World!</h1>", template.render.strip
+      assert_equal "<h1 id=\"label-Hello+World%21\">Hello World!</h1>", template.render.strip
     end
 
     test "can be rendered more than once" do
       template = Tilt::RDocTemplate.new { |t| "= Hello World!" }
-      3.times { assert_equal "<h1>Hello World!</h1>", template.render.strip }
+      3.times { assert_equal "<h1 id=\"label-Hello+World%21\">Hello World!</h1>", template.render.strip }
     end
   end
 rescue LoadError => boom
