--- lib/syntax/lang/yaml.rb.orig	2012-01-31 15:07:34.533346064 +0100
+++ lib/syntax/lang/yaml.rb	2012-01-31 15:06:26.365562358 +0100
@@ -62,7 +62,7 @@
             start_group :string
             loop do
               line = check_until(/[\n\r]|\Z/)
-              break if line.nil?
+              break if line.nil? or line.empty?
               if line.chomp.length > 0
                 this_indent = line.chomp.match( /^\s*/ )[0]
                 break if this_indent.length < indent.length
