diff -ru ipaddress-0.8.0/test/ipaddress/ipv4_test.rb ipaddress-0.8.0.new/test/ipaddress/ipv4_test.rb
--- ipaddress-0.8.0/test/ipaddress/ipv4_test.rb	2015-12-22 12:34:10.810501416 +0000
+++ ipaddress-0.8.0.new/test/ipaddress/ipv4_test.rb	2015-12-22 12:33:08.921287474 +0000
@@ -109,7 +109,11 @@
   end
 
   def test_method_data
-    assert_equal "\254\020\n\001", @ip.data
+    if RUBY_VERSION < "2.0"
+      assert_equal "\254\020\n\001", @ip.data
+    else
+      assert_equal "\xAC\x10\n\x01".b, @ip.data
+    end
   end
   
   def test_method_to_string
diff -ru ipaddress-0.8.0/test/ipaddress/ipv6_test.rb ipaddress-0.8.0.new/test/ipaddress/ipv6_test.rb
--- ipaddress-0.8.0/test/ipaddress/ipv6_test.rb	2015-12-22 12:34:10.811501420 +0000
+++ ipaddress-0.8.0.new/test/ipaddress/ipv6_test.rb	2015-12-22 12:33:08.921287474 +0000
@@ -180,7 +180,11 @@
   end
   
   def test_method_data
-    str = " \001\r\270\000\000\000\000\000\b\b\000 \fAz"
+    if RUBY_VERSION < "2.0"
+      str = " \001\r\270\000\000\000\000\000\b\b\000 \fAz"
+    else
+      str = " \x01\r\xB8\x00\x00\x00\x00\x00\b\b\x00 \fAz".b
+    end
     assert_equal str, @ip.data
   end
 
