--- rkerberos-0.2.0/ext/rkerberos/policy.c.gcc8	2024-01-01 00:00:00.000000000 +0000
+++ rkerberos-0.2.0/ext/rkerberos/policy.c	2024-01-01 00:00:00.000000000 +0000
@@ -59,7 +59,7 @@ static VALUE rkadm5_policy_init(VALUE se
 
   Check_Type(v_options, T_HASH);
 
-  if(RTEST(rb_funcall(v_options, rb_intern("empty?"), 0, 0)))
+  if(RTEST(rb_funcall(v_options, rb_intern("empty?"), 0)))
     rb_raise(rb_eArgError, "no policy options provided");
 
   v_name       = rb_hash_aref2(v_options, rb_str_new_cstr("name"));
