diff --git a/lib/phusion_passenger/native_support.rb b/lib/phusion_passenger/native_support.rb
index 994c379..d2b84fc 100644
--- a/lib/phusion_passenger/native_support.rb
+++ b/lib/phusion_passenger/native_support.rb
@@ -113,7 +113,7 @@ private
 	end
 	
 	def load_from_load_path
-		require 'passenger_native_support'
+		require 'native/passenger_native_support'
 		return true
 	rescue LoadError
 		return false
