--- a/spec/backends/swiftiply_client_spec.rb
+++ b/spec/backends/swiftiply_client_spec.rb
@@ -3,7 +3,7 @@
 describe Backends::SwiftiplyClient do
   before do
     @backend = Backends::SwiftiplyClient.new('0.0.0.0', 3333)
-    @backend.server = mock('server', :null_object => true)
+    @backend.server = mock('server').as_null_object
   end
   
   it "should connect" do
@@ -26,7 +26,7 @@
   before do
     @connection = SwiftiplyConnection.new(nil)
     @connection.backend = Backends::SwiftiplyClient.new('0.0.0.0', 3333)
-    @connection.backend.server = mock('server', :null_object => true)
+    @connection.backend.server = mock('server').as_null_object
   end
   
   it do
--- a/spec/connection_spec.rb
+++ b/spec/connection_spec.rb
@@ -2,7 +2,7 @@
 
 describe Connection do
   before do
-    @connection = Connection.new(mock('EM', :null_object => true))
+    @connection = Connection.new(mock('EM').as_null_object)
     @connection.post_init
     @connection.backend = mock("backend", :ssl? => false)
     @connection.app = proc do |env|
