From 8c85abf355cfdb81bd87a0e1d11dc3876e71a5c6 Mon Sep 17 00:00:00 2001
From: Travis Chase <travis@travischase.me>
Date: Wed, 7 Nov 2018 08:23:48 -0600
Subject: [PATCH] Update to use either sass or sassc gem

---
 lib/font-awesome-sass.rb | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/lib/font-awesome-sass.rb b/lib/font-awesome-sass.rb
index 1747cd5..7c63bb2 100644
--- a/lib/font-awesome-sass.rb
+++ b/lib/font-awesome-sass.rb
@@ -2,15 +2,15 @@ module FontAwesome
   module Sass
     class << self
       def load!
-        register_compass_extension if compass?
-
         if rails?
           register_rails_engine
         elsif sprockets?
           register_sprockets
+        elsif compass?
+          register_compass_extension
+        elsif deprecated_sass?
+          ::Sass.load_paths << stylesheets_path
         end
-
-        configure_sass
       end
 
       # Paths
@@ -42,14 +42,13 @@ module FontAwesome
         defined?(::Rails)
       end
 
-      private
-
-      def configure_sass
-        require 'sass'
-
-        ::Sass.load_paths << stylesheets_path
+      def deprecated_sass?
+        # The deprecated `sass` gem:
+        defined?(::Sass) && ::Sass.respond_to?(:load_paths)
       end
 
+      private
+
       def register_compass_extension
         require 'font_awesome/sass/version'
 
-- 
2.37.1

