--- a/config/settings.yml.example
+++ b/config/settings.yml.example
@@ -1,4 +1,7 @@
 ---
+#replace default location of "settings.d"
+:settings_directory: /etc/foreman-proxy/settings.d
+
 # SSL Setup
 # If enabled, all communication would be verified via SSL
 # NOTE that both certificates need to be signed by the same CA in order for this to work
@@ -21,7 +24,7 @@
 #- foreman.prod.domain
 #- foreman.dev.domain
 #to deny access to all hosts use:
-#:trusted_hosts: []
+:trusted_hosts: []
 
 # verify a DNS reverse lookup against it's forward lookup
 # 1.1.1.1 -> foreman.mycompany.com -> 1.1.1.1
@@ -38,16 +41,16 @@
 #:foreman_ssl_key: ssl/private_keys/fqdn.pem
 
 # by default smart_proxy runs in the foreground. To enable running as a daemon, uncomment 'daemon' setting
-#:daemon: true
+:daemon: true
 # Only used when 'daemon' is set to true.
 # Uncomment and modify if you want to change the default pid file '/var/run/foreman-proxy/foreman-proxy.pid'
 #:daemon_pid: /var/run/foreman-proxy/foreman-proxy.pid
 
 # host and ports configuration
 # an array of interfaces to bind ports to (possible values: *, localhost, 0.0.0.0)
-#:bind_host: ['*']
+:bind_host: ['localhost']
 # http is disabled by default. To enable, uncomment 'http_port' setting
-#:http_port: 8000
+:http_port: 8000
 # https is enabled if certificate, CA certificate, and private key are present in locations specifed by
 # ssl_certificate, ssl_ca_file, and ssl_private_key correspondingly
 # default values for https_port is 8443
