From 138064cccf7d010db03fecd08a2626dbd2629e39 Mon Sep 17 00:00:00 2001
From: Tomas Strachota <tstrachota@redhat.com>
Date: Thu, 14 Aug 2014 12:30:03 +0000
Subject: [PATCH] Refs #7083 - adding system locale domain

---
 lib/hammer_cli/i18n.rb |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/lib/hammer_cli/i18n.rb b/lib/hammer_cli/i18n.rb
index a40746f..c1e2e17 100644
--- a/lib/hammer_cli/i18n.rb
+++ b/lib/hammer_cli/i18n.rb
@@ -102,6 +102,14 @@ module HammerCLI
 
     end
 
+    class SystemLocaleDomain < LocaleDomain
+
+      def locale_dir
+        '/usr/share/locale'
+      end
+
+    end
+
 
     def self.locale
       lang_variant = Locale.current.to_simple.to_str
@@ -145,4 +153,5 @@ include FastGettext::Translation
 include HammerCLI::I18n::AllDomains
 
 HammerCLI::I18n.add_domain(HammerCLI::I18n::LocaleDomain.new)
+HammerCLI::I18n.add_domain(HammerCLI::I18n::SystemLocaleDomain.new)
 
-- 
1.7.1

