diff -up ./build/packaging.rb.nosign ./build/packaging.rb
--- ./build/packaging.rb.nosign	2013-05-28 18:33:10.117895835 +0200
+++ ./build/packaging.rb	2013-05-28 18:33:32.246157763 +0200
@@ -88,7 +88,7 @@ end
 
 task 'package:gem' => Packaging::PREGENERATED_FILES do
 	require 'phusion_passenger'
-	sh "gem build #{PhusionPassenger::PACKAGE_NAME}.gemspec --sign --key 0x0A212A8C"
+	sh "gem build #{PhusionPassenger::PACKAGE_NAME}.gemspec"
 	sh "mkdir -p pkg"
 	sh "mv #{PhusionPassenger::PACKAGE_NAME}-#{PhusionPassenger::VERSION_STRING}.gem pkg/"
 end
