diff -urp passenger-release-4.0.18.orig/lib/phusion_passenger/packaging.rb passenger-release-4.0.18/lib/phusion_passenger/packaging.rb
--- passenger-release-4.0.18.orig/lib/phusion_passenger/packaging.rb	2013-09-17 13:47:54.000000000 -0500
+++ passenger-release-4.0.18/lib/phusion_passenger/packaging.rb	2013-09-24 14:42:54.126310781 -0500
@@ -36,7 +36,7 @@ module Packaging
 	# Files that must be generated before packaging.
 	PREGENERATED_FILES = [
 		'ext/common/Constants.h',
-		'doc/Packaging.html'
+		'doc/Packaging.txt.md'
 	] + ASCII_DOCS
 	
 	USER_EXECUTABLES = [
