Description: Use distro provided inventory files for intersphinx
Forwarded: not-needed
Author: Sakirnth Nagarasa <sakirnth@gmail.com>
Last-Update: 2021-11-16
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -162,7 +162,7 @@
 # -- Options for intersphinx extension ---------------------------------------
 
 # Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {'https://docs.python.org/': None}
+intersphinx_mapping = {'/usr/share/doc/python3-doc/html': None}
 
 # -- Options for todo extension ----------------------------------------------
 
