From 56f26e01049ddfad13ded0edbbf5c60a3246720d Mon Sep 17 00:00:00 2001
From: Link Dupont <link@sub-pop.net>
Date: Fri, 14 May 2021 14:55:06 -0400
Subject: [PATCH] build: Remove the Makefile preamble

Signed-off-by: Link Dupont <link@sub-pop.net>
---
 Makefile | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/Makefile b/Makefile
index 0bc0efc..f1f4380 100644
--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,3 @@
-SHELL := bash
-.ONESHELL:
-.SHELLFLAGS := -eu -o pipefail -c
-.DELETE_ON_ERROR:
-MAKEFLAGS += --warn-undefined-variables
-MAKEFLAGS += --no-builtin-rules
-.POSIX:
-.SUFFIXES:
-
 # Project variables
 
 # Used as a prefix to binary names. Cannot contain spaces.
