From b8e4bbd37ef3ab902adfca3f0b9fbc8059f4b0fa Mon Sep 17 00:00:00 2001
From: Adam Ruzicka <aruzicka@redhat.com>
Date: Tue, 19 Sep 2023 16:17:07 +0200
Subject: [PATCH] Propagate FOREMAN_REX_WORKDIR to workers

---
 cmd/yggd/main.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmd/yggd/main.go b/cmd/yggd/main.go
index 5c8aebc..721ed16 100644
--- a/cmd/yggd/main.go
+++ b/cmd/yggd/main.go
@@ -377,6 +377,10 @@ func main() {
 			"YGG_SOCKET_ADDR=unix:" + c.String("socket-addr"),
 			"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
 		}
+		foremanYggWorkerDir, foremanYggWorkerDirP := os.LookupEnv("FOREMAN_YGG_WORKER_WORKDIR")
+		if foremanYggWorkerDirP {
+			env = append(env, "FOREMAN_YGG_WORKER_WORKDIR="+foremanYggWorkerDir)
+		}
 		for _, info := range fileInfos {
 			if strings.HasSuffix(info.Name(), "worker") {
 				if ExcludeWorkers[info.Name()] {
-- 
2.39.3

