From 38cdd49ea90638f39c1a9bc6166f7b1f562deaba Mon Sep 17 00:00:00 2001
From: Dominic Cleal <dcleal@redhat.com>
Date: Tue, 20 Aug 2013 18:53:34 +0100
Subject: [PATCH] Find build_iso.sh script from __FILE__ not bundler

---
 lib/tasks.rake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/tasks.rake b/lib/tasks.rake
index 88bc124..0fb78be 100644
--- a/lib/tasks.rake
+++ b/lib/tasks.rake
@@ -43,7 +43,7 @@ namespace :discovery do
       exit 1
     end
 
-    script=`bundle show foreman_discovery`.strip + '/extra/build_iso.sh'
+    script = File.join(File.dirname(__FILE__), '..', 'extra', 'build_iso.sh')
     builddir=''
     begin
       status = Open4::popen4("#{prefix} #{script} #{mode}") do |pid, stdin, stdout, stderr|
-- 
1.8.1.6

