diff --git a/netbox-client-ruby.gemspec b/netbox-client-ruby.gemspec
--- netbox-client-ruby.gemspec
+++ netbox-client-ruby.gemspec
@@ -28,7 +28,6 @@
   spec.add_runtime_dependency 'faraday-detailed_logger', '~> 2.1'
   spec.add_runtime_dependency 'faraday_middleware', '~> 0.11'
   spec.add_runtime_dependency 'ipaddress', '~> 0.8', '>= 0.8.3'
-  spec.add_runtime_dependency 'openssl', '~> 2.0', '>= 2.0.5'

   spec.add_development_dependency 'bundler', '~> 2.1'
   spec.add_development_dependency 'pry', '~> 0.10'
