Description: Fix require relative to test dir.
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2012-05-15
--- a/test/ts_all.rb
+++ b/test/ts_all.rb
@@ -7,14 +7,14 @@
 
 require "test/unit"
 
-require "tc_csv_parsing"
-require "tc_features"
-require "tc_interface"
-require "tc_csv_writing"
-require "tc_speed"
-require "tc_data_converters"
-require "tc_row"
-require "tc_table"
-require "tc_headers"
-require "tc_serialization"
-require "tc_encodings"
+require "./test/tc_csv_parsing"
+require "./test/tc_features"
+require "./test/tc_interface"
+require "./test/tc_csv_writing"
+#require "./test/tc_speed"
+require "./test/tc_data_converters"
+require "./test/tc_row"
+require "./test/tc_table"
+require "./test/tc_headers"
+require "./test/tc_serialization"
+require "./test/tc_encodings"
