Description: Remove shebangs in library files, rename in examples.
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2012-05-15
--- a/lib/faster_csv.rb
+++ b/lib/faster_csv.rb
@@ -1,5 +1,3 @@
-#!/usr/local/bin/ruby -w
-
 # = faster_csv.rb -- Faster CSV Reading and Writing
 #
 #  Created by James Edward Gray II on 2005-10-31.
--- a/lib/fastercsv.rb
+++ b/lib/fastercsv.rb
@@ -1,5 +1,3 @@
-#!/usr/local/bin/ruby -w
-
 # fastercsv.rb
 #
 #  Created by James Edward Gray II on 2006-04-01.
--- a/examples/csv_converters.rb
+++ b/examples/csv_converters.rb
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby -w
+#!/usr/bin/env ruby
 
 # csv_converters.rb
 #
--- a/examples/csv_filter.rb
+++ b/examples/csv_filter.rb
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby -w
+#!/usr/bin/env ruby
 
 # = csv_filter.rb -- Faster CSV Reading and Writing
 #
--- a/examples/csv_reading.rb
+++ b/examples/csv_reading.rb
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby -w
+#!/usr/bin/env ruby
 
 # csv_reading.rb
 #
--- a/examples/csv_table.rb
+++ b/examples/csv_table.rb
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby -w
+#!/usr/bin/env ruby
 
 # csv_table.rb
 #
--- a/examples/csv_writing.rb
+++ b/examples/csv_writing.rb
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby -w
+#!/usr/bin/env ruby
 
 # csv_rails_import.rb
 #
--- a/examples/shortcut_interface.rb
+++ b/examples/shortcut_interface.rb
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby -w
+#!/usr/bin/env ruby
 
 # shortcut_interface.rb
 #
