From 55a38b238c6d455ee5f521907625604dacacf8ec Mon Sep 17 00:00:00 2001
From: Tomas Strachota <tstrachota@redhat.com>
Date: Thu, 14 Aug 2014 12:29:51 +0000
Subject: [PATCH] Refs #7083 - adding system locale domain

---
 lib/hammer_cli_foreman/i18n.rb |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/lib/hammer_cli_foreman/i18n.rb b/lib/hammer_cli_foreman/i18n.rb
index ec52882..db19d4e 100644
--- a/lib/hammer_cli_foreman/i18n.rb
+++ b/lib/hammer_cli_foreman/i18n.rb
@@ -18,7 +18,16 @@ module HammerCLIForeman
       end
     end

+    class SystemLocaleDomain < LocaleDomain
+
+      def locale_dir
+        '/usr/share/locale'
+      end
+
+    end
+
   end
 end

 HammerCLI::I18n.add_domain(HammerCLIForeman::I18n::LocaleDomain.new)
+HammerCLI::I18n.add_domain(HammerCLIForeman::I18n::SystemLocaleDomain.new)
--
1.7.1

