--- test/active_record_unit.rb.orig	2011-01-10 23:31:54.197506001 -0500
+++ test/active_record_unit.rb	2011-01-11 00:09:09.152505905 -0500
@@ -13,12 +13,12 @@ end
 # Try to grab AR
 unless defined?(ActiveRecord) && defined?(Fixtures)
   begin
-    PATH_TO_AR = "#{File.dirname(__FILE__)}/../../activerecord/lib"
-    raise LoadError, "#{PATH_TO_AR} doesn't exist" unless File.directory?(PATH_TO_AR)
-    $LOAD_PATH.unshift PATH_TO_AR
+    #PATH_TO_AR = "#{File.dirname(__FILE__)}/../../activerecord/lib"
+    #raise LoadError, "#{PATH_TO_AR} doesn't exist" unless File.directory?(PATH_TO_AR)
+    #$LOAD_PATH.unshift PATH_TO_AR
     require 'active_record'
   rescue LoadError => e
-    $stderr.print "Failed to load Active Record. Skipping Active Record assertion tests: #{e}"
+    #$stderr.print "Failed to load Active Record. Skipping Active Record assertion tests: #{e}"
     ActiveRecordTestConnector.able_to_connect = false
   end
 end
