--- TMP/lib/hoe/publish.rb.dep	1970-01-01 09:00:00.000000000 +0900
+++ TMP/lib/hoe/publish.rb	2011-09-09 03:08:08.000000000 +0900
@@ -93,11 +93,13 @@

       begin
         require 'rdoc/task'
+        rdoctask = RDoc::Task
       rescue LoadError
         require 'rake/rdoctask'
+        rdoctask = Rake::RDocTask
       end

-      RDoc::Task.new(:docs) do |rd|
+      rdoctask.new(:docs) do |rd|
         rd.main = readme_file
         rd.options << '-d' if (`which dot` =~ /\/dot/) unless
           ENV['NODOT'] || Hoe::WINDOZE
