diff --git a/lib/configure.js b/lib/configure.js
index 564564e..d010063 100644
--- a/lib/configure.js
+++ b/lib/configure.js
@@ -37,7 +37,7 @@ function configure (gyp, argv, callback) {

     if (gyp.opts.nodedir) {
       // --nodedir was specified. use that for the dev files
-      nodeDir = gyp.opts.nodedir.replace(/^~/, os.homedir())
+      nodeDir = '/opt/rh/rh-nodejs10/root/usr/share/node'

       log.verbose('get node dir', 'compiling against specified --nodedir dev files: %s', nodeDir)
       createBuildDir()
@@ -48,7 +48,10 @@ function configure (gyp, argv, callback) {
         log.verbose('get node dir', 'compiling against --target node version: %s', release.version)
       } else {
         // if no --target was specified then use the current host node version
+        // if no --target was specified then use RPM installed headers
+
         log.verbose('get node dir', 'no --target version specified, falling back to host node version: %s', release.version)
+        log.verbose('get node dir', 'no --target version specified, falling back to RPM installed headers')
       }

       if (!release.semver) {
@@ -65,7 +65,7 @@ function configure (gyp, argv, callback) {
           return callback(err)
         }
         log.verbose('get node dir', 'target node version installed:', release.versionDir)
-        nodeDir = path.resolve(gyp.devDir, release.versionDir)
+        nodeDir = '/opt/rh/rh-nodejs10/root/usr/share/node'
         createBuildDir()
       })
     }
@@ -281,9 +284,10 @@ function configure (gyp, argv, callback) {
     }

     // this logic ported from the old `gyp_addon` python file
-    var gypScript = path.resolve(__dirname, '..', 'gyp', 'gyp_main.py')
-    var addonGypi = path.resolve(__dirname, '..', 'addon.gypi')
-    var commonGypi = path.resolve(nodeDir, 'include/node/common.gypi')
+    var gypScript = '/usr/bin/gyp'
+    var addonGypiFile = gyp.opts.target || gyp.opts.nodeDir ? 'addon.gypi' : 'addon-rpm.gypi'
+    var addonGypi = path.resolve(__dirname, '..', addonGypiFile)
+    var commonGypi = path.resolve(nodeDir, '/opt/rh/rh-nodejs10/root/usr/include/node/common.gypi')
     fs.stat(commonGypi, function (err) {
       if (err) {
         commonGypi = path.resolve(nodeDir, 'common.gypi')
@@ -320,7 +324,7 @@ function configure (gyp, argv, callback) {
       argv.push('-Dnode_engine=' +
         (gyp.opts.node_engine || process.jsEngine || 'v8'))
       argv.push('--depth=.')
-      argv.push('--no-parallel')
+      //argv.push('--no-parallel')

       // tell gyp to write the Makefile/Solution files into output_dir
       argv.push('--generator-output', outputDir)
--
2.24.1
