From 719986c444f6cd9d2a9dc4109b95d036a6494c38 Mon Sep 17 00:00:00 2001
From: Dirk Goetz <dirk.goetz@netways.de>
Date: Thu, 30 Apr 2020 14:26:32 +0200
Subject: [PATCH] allow to_sentence for array

---
 lib/safemode/core_jails.rb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/safemode/core_jails.rb b/lib/safemode/core_jails.rb
index ec875fe..bf8e395 100644
--- a/lib/safemode/core_jails.rb
+++ b/lib/safemode/core_jails.rb
@@ -51,8 +51,8 @@ def core_jail_class_methods(klass)
                     indexes indices inject insert join last length map map!
                     nitems pop push present? rassoc reject reject! reverse
                     reverse! reverse_each rindex select shift size slice
-                    slice! sort sort! transpose uniq uniq! unshift values_at
-                    zip),
+                    slice! sort sort! transpose to_sentence uniq uniq! unshift
+                    values_at zip),
 
     'Bignum'     => %w(abs blank? ceil chr coerce div divmod downto floor hash
                     integer? modulo next nonzero? present? quo remainder round
