diff -Nrwu certifi-2020.6.20.orig/certifi/core.py certifi-2020.6.20/certifi/core.py
--- certifi-2020.6.20.orig/certifi/core.py	2020-06-07 11:35:28.000000000 +0200
+++ certifi-2020.6.20/certifi/core.py	2020-07-20 14:28:41.063958002 +0200
@@ -9,6 +9,7 @@
 import os
 
 try:
+    raise ImportError  # force fallback
     from importlib.resources import path as get_path, read_text
 
     _CACERT_CTX = None
@@ -51,9 +52,7 @@
     # If we don't have importlib.resources, then we will just do the old logic
     # of assuming we're on the filesystem and munge the path directly.
     def where():
-        f = os.path.dirname(__file__)
-
-        return os.path.join(f, "cacert.pem")
+        return '/etc/pki/tls/certs/ca-bundle.crt'
 
 
 def contents():
