From 86fa3447643e890ffe26fe00ad57abb12565311e Mon Sep 17 00:00:00 2001
From: Dominic Cleal <dcleal@redhat.com>
Date: Tue, 11 Jun 2013 17:46:19 +0100
Subject: [PATCH] Fix missing runtime info properties

---
 test/test_deserialization.rb | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test/test_deserialization.rb b/test/test_deserialization.rb
index a3b56c8..2cb8d85 100644
--- a/test/test_deserialization.rb
+++ b/test/test_deserialization.rb
@@ -252,11 +252,14 @@ end
       :connectionState => "connected",
       :dynamicProperty => [],
       :faultToleranceState => "notConfigured",
+      :featureMask => [],
+      :featureRequirement => [],
       :host => VIM::HostSystem(nil, "host-32"),
       :maxCpuUsage => 5612,
       :maxMemoryUsage => 3072,
       :memoryOverhead => 128671744,
       :numMksConnections => 1,
+      :offlineFeatureRequirement => [],
       :powerState => "poweredOn",
       :recordReplayState => "inactive",
       :suspendInterval => 0,
-- 
1.8.1.4

