From 8f3827aaa23bf4dd67f5fd671076b8d3957772b8 Mon Sep 17 00:00:00 2001
From: Dominic Cleal <dcleal@redhat.com>
Date: Tue, 11 Jun 2013 16:16:40 +0100
Subject: [PATCH] Fix assumption that the time test is always in -0800

---
 test/test_serialization.rb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/test_serialization.rb b/test/test_serialization.rb
index c036e4e..b107af0 100644
--- a/test/test_serialization.rb
+++ b/test/test_serialization.rb
@@ -287,9 +287,9 @@ class SerializationTest < Test::Unit::TestCase
   end
 
   def test_time
-    obj = Time.at DateTime.new(2011, 11, 16, 13, 36, 8, Rational(-8,24)).strftime("%s").to_i
+    obj = Time.at(DateTime.new(2011, 11, 16, 13, 36, 8, Rational(-8,24)).strftime("%s").to_i).getgm
     check <<-EOS, obj, 'xsd:dateTime', false
-<root>2011-11-16T13:36:08-08:00</root>
+<root>2011-11-16T21:36:08Z</root>
     EOS
   end
 
-- 
1.8.1.4

