--- specifications/rack-mount-0.7.1.gemspec.orig
+++ specifications/rack-mount-0.7.1.gemspec
@@ -23,15 +23,18 @@
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
       s.add_runtime_dependency(%q<rack>, [">= 1.0.0"])
+      s.add_runtime_dependency(%q<regin>, [">= 0.3.7"])
       s.add_development_dependency(%q<racc>, [">= 0"])
       s.add_development_dependency(%q<rexical>, [">= 0"])
     else
       s.add_dependency(%q<rack>, [">= 1.0.0"])
+      s.add_dependency(%q<regin>, [">= 0.3.7"])
       s.add_dependency(%q<racc>, [">= 0"])
       s.add_dependency(%q<rexical>, [">= 0"])
     end
   else
     s.add_dependency(%q<rack>, [">= 1.0.0"])
+    s.add_dependency(%q<regin>, [">= 0.3.7"])
     s.add_dependency(%q<racc>, [">= 0"])
     s.add_dependency(%q<rexical>, [">= 0"])
   end

