--- a/lib/unicode/display_width/constants.rb
+++ b/lib/unicode/display_width/constants.rb
@@ -2,7 +2,7 @@
   module DisplayWidth
     VERSION = '1.1.1'
     UNICODE_VERSION = "9.0.0".freeze
-    DATA_DIRECTORY = File.expand_path(File.dirname(__FILE__) + '/../../../data/').freeze
+    DATA_DIRECTORY = '/usr/share/ruby-unicode-display-width'.freeze
     INDEX_FILENAME = (DATA_DIRECTORY + '/display_width.marshal.gz').freeze
   end
 end
