From: Dmitry Borodaenko <angdraug@debian.org>
Date: Sun, 18 Sep 2011 13:24:18 +0300
Subject: fix rspec requires

---
 spec/spec_helper.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/spec/spec_helper.rb b/spec/spec_helper.rb
index 23b4387..f2cc4a5 100644
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -1,6 +1,7 @@
 # $VERBOSE = true # ignore complaints in spec files
 
 # ---- requirements
+require 'rspec'
 require 'fast_gettext'
 require 'active_record'
 
